app/process/watcher/prometheus/watcher.go

```diff
@@ -225,117 +254,149 @@ func (pw Watcher) getAccount(accountId string) *model.AccountsResponse {
-
-				wrappedInstance, e := wtoken.NewWtoken(common.HexToAddress(asset), evm)
-				if e != nil {
-					panic(e)
-				}
-
-				pw.setEvmAssetSupplyMetric(wrappedInstance, asset)
-				pw.setEvmAssetBalanceMetric(wrappedInstance, asset, address)
+func (pw Watcher) setAssetsMetrics(bridgeAccount *model.AccountsResponse) {
+	nativeToWrapped := pw.configuration.Bridge.Assets.GetNativeToWrapped()
+	for nativeNetworkId, nativeMap := range nativeToWrapped {
+		for nativeAsset, wrappedMap := range nativeMap {
+			// set native assets balance
+			pw.setAssetMetric(nativeNetworkId, nativeAsset, bridgeAccount, true)
+			for wrappedNetworkId, wrappedAsset := range wrappedMap {
+				//set wrapped assets total supply
+				pw.setAssetMetric(wrappedNetworkId, wrappedAsset, bridgeAccount, false)
 			}
 		}
 	}
 }
 
-func (pw Watcher) setBridgeAccAssetsMetrics(bridgeAcc *model.AccountsResponse) {
+func (pw Watcher) setAssetMetric(networkId int64,
+	assetAddress string,
+	bridgeAccount *model.AccountsResponse,
+	isNative bool,
+) {
 	if !pw.prometheusService.GetIsMonitoringEnabled() {
 		return
 	}
 
-	for _, token := range bridgeAcc.Balance.Tokens {
-		metric := pw.prometheusService.GetGauge(pw.bridgeAccAssetsMetrics[token.TokenID])
-		if metric == nil {
-			log.Infof("Skip metrics for Bridge Account asset with ID [%s]", token.TokenID)
-		} else {
-			metric.Set(float64(token.Balance))
-			log.Infof("The Bridge Account asset with ID [%s] has Balance = %f", token.TokenID, float64(token.Balance))
+	if assetAddress != constants.Hbar { // skip HBAR
+		assetMetric := pw.prometheusService.GetGauge(pw.assetsMetrics[networkId][assetAddress])
+		value := pw.getAssetMetricValue(networkId, assetAddress, bridgeAccount, isNative)
+		logStr := constants.SupplyAssetMetricsHelpPrefix
+		if isNative {
+			logStr = constants.BalanceAssetMetricHelpPrefix
 		}
+		setMetric(assetMetric, assetAddress, value, logStr)
 	}
 }
 
-func (pw Watcher) setHederaAssetSupplyMetric(asset string, token *model.TokenResponse) {
-	if !pw.prometheusService.GetIsMonitoringEnabled() {
-		return
+func (pw Watcher) getAssetMetricValue(
+	networkId int64,
+	assetAddress string,
+	bridgeAccount *model.AccountsResponse,
+	isNative bool,
+) (value float64) {
+	var (
+		evmAssetInstance *wtoken.Wtoken
+		decimal          uint8
+	)
+	if networkId != constants.HederaNetworkId { // EVM
+		evm := pw.EVMClients[networkId].GetClient()
+		wrappedTokenInstance, e := wtoken.NewWtoken(common.HexToAddress(assetAddress), evm)
+		if e != nil {
+			panic(e)
+		}
+		evmAssetInstance = wrappedTokenInstance
+		dec, e := evmAssetInstance.Decimals(&bind.CallOpts{})
+		if e != nil {
+			panic(e)
+		}
+		decimal = dec
 	}
 
-	totalSupplyMetric := pw.prometheusService.GetGauge(pw.supplyAssetsMetrics[asset])
-	pw.setHederaAssetMetric(totalSupplyMetric, token.TotalSupply, asset)
+	if networkId == constants.HederaNetworkId && isNative { // Hedera native balance
```

Comment:
You can use 1 `if` statement for the `networkId` and inside to use another `if` statement for the `tokenType` to avoid duplication of checks. As otherwise if it doesn't enter the first statement it will check again the `networkId` for each other check until it gets to the one for which is the case. This way it won't do unnecessary operations and improves readability.